File: 0283-move-zeroes/0283-move-zeroes.java
class Solution {
    public void moveZeroes(int[] nums) {
        int l=0,r=nums.length-1;
        for(int i=0;i<r+1;i++){
            if(nums[i]!=0){
                int temp=nums[i];
                nums[i]=nums[l];
                nums[l]=temp;
                l=l+1;

            }
        }
      //  return nums;
    }
}

File: 0283-move-zeroes/0283-move-zeroes.cpp
class Solution {
public:
    void moveZeroes(vector<int>& nums) {
        int write=0;
        for(int read=0;read<nums.size();read++){
            if(nums[read]!=0){
                swap(nums[write],nums[read]);
                write++;
            }
        }
       // return nums;
    }
};